starlette/testclient.py

```diff
@@ -315,6 +316,7 @@ def __enter__(self) -> "WebSocketTestSession":
             self.exit_stack.close()
             raise
         self.accepted_subprotocol = message.get("subprotocol", None)
+        self.extra_headers = message.get("headers", None)
```

Comment:
This should probably be `message.get("headers", [])` with #1422

Reply:
I think this part is ok. When passing `None` you'll need to check `assert websocket.extra_headers is None` but with that change:

```python
def test_additional_headers(test_client_factory):
     def app(scope):
         async def asgi(receive, send):
             websocket = WebSocket(scope, receive=receive, send=send)
             await websocket.accept(headers=None)
             await websocket.close()

         return asgi

     client = test_client_factory(app)
     with client.websocket_connect("/") as websocket:
         assert websocket.extra_headers == []
```

Am I missing something?

Reply:
You're not. I don't have strong feelings for this. :+1: